app/uk/gov/hmrc/tai/connectors/CacheConnector.scala

```diff
@@ -16,36 +16,22 @@
 
 package uk.gov.hmrc.tai.connectors
 
+import cats.data.OptionT
+import cats.implicits._
 import com.google.inject.{Inject, Singleton}
-import play.Logger
 import play.api.Configuration
-import play.api.libs.json.{JsResultException, JsValue, Json, Reads, Writes}
-import play.modules.reactivemongo.ReactiveMongoComponent
-import uk.gov.hmrc.cache.repository.CacheMongoRepository
+import play.api.libs.json._
 import uk.gov.hmrc.crypto.json.{JsonDecryptor, JsonEncryptor}
 import uk.gov.hmrc.crypto.{ApplicationCrypto, CompositeSymmetricCrypto, Protected}
 import uk.gov.hmrc.tai.config.MongoConfig
 import uk.gov.hmrc.tai.model.nps2.MongoFormatter
-import cats.implicits._
-import cats.data.OptionT
-import uk.gov.hmrc.cache.model.Cache
 
 import scala.concurrent.{ExecutionContext, Future}
 
-@Singleton
-class TaiCacheRepository @Inject()(mongo: ReactiveMongoComponent, mongoConfig: MongoConfig)(
-  implicit ec: ExecutionContext)
-  extends CacheMongoRepository("TAI", mongoConfig.mongoTTL)(mongo.mongoConnector.db, ec)
-
-class TaiCacheRepositoryUpdateIncome @Inject()(mongo: ReactiveMongoComponent, mongoConfig: MongoConfig)(
```

Comment:
Can we do something like this rather than creating two new files, and if we do need to create 2 new files should they be in the "repository" package and not the connector one

Reply:
Repository package is already being used for external calls so did not move it there. But we can have another package _cacherepository_  if that helps

Reply:
I think with just 2 repositories we can have them as concrete implementation with different TTLs. But we can discuss this. I have added it as potential refactoring based on whatever we decide

https://jira.tools.tax.service.gov.uk/browse/PAYEE-944